src/service_objects/mina/pieces_tree_serializer.ts

```diff
@@ -0,0 +1,70 @@
+import { Field, UInt32 } from 'snarkyjs';
+import {
+  Piece,
+  PiecesMerkleTree,
+  Position,
+  Unit,
+  UnitStats
+} from 'mina-arena-contracts';
+
+import * as Models from '../../models/index.js';
+import { Transaction } from 'sequelize';
+
+/**
+ * Given a Game ID, find all of the Game Pieces and serialize them into a PiecesMerkleTree
+ */
+
+// export default async function serializePiecesTree(gameId: string | number, transaction?: Transaction): Promise<PiecesMerkleTree> {
+//   const game = await Models.Game.findByPk(gameId, { transaction });
+//   if (game.status != 'inProgress') throw new Error(`Game ${game.id} is not in progress, status: ${game.status}`);
+
+//   const gamePieces = await Models.GamePiece.findAll({
+//     where: { gameId },
+//     include: [
+//       {
+//         model: Models.GamePlayer,
+//         as: 'gamePlayer',
+//       },
+//       {
+//         model: Models.PlayerUnit,
+//         as: 'playerUnit',
+//         include: [
+//           {
+//             model: Models.Unit,
+//             as: 'unit',
+//           }
+//         ]
+//       }
+//     ],
+//   });
+
+//   const piecesTree = new PiecesMerkleTree();
+//   gamePieces.forEach((gamePiece) => {
+//     const snarkyUnit = new Unit({
+//       stats: new UnitStats({
+//         health: UInt32.from(gamePiece.health),
+//         movement: UInt32.from(gamePiece.playerUnit.unit.movementSpeed), // TODO: How does this work?  Currently getting `Property 'unit' does not exist on type '() => Promise<PlayerUnit>'`
```

Comment:
Could use help on the todo here

Reply:
You might need to do `gamePiece.playerUnit().unit().movementSpeed`. If one doesn't exist, we could also add a `unit()` helper function to the GamePiece class to make this query more efficient, doesn't matter for now though.

Reply:
Yup, got this working by calling the methods and awaiting them.